src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1647,6 +1665,20 @@ public async Task AutoAddSeatsAsync(Organization organization, int seatsToAdd, D
 
     [Obsolete("IDeleteOrganizationUserCommand should be used instead. To be removed by EC-607.")]
     public async Task DeleteUserAsync(Guid organizationId, Guid organizationUserId, Guid? deletingUserId)
+    {
+        var orgUser = await RepositoryDeleteUserAsync(organizationId, organizationUserId, deletingUserId);
+        await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Removed);
+    }
+
+    [Obsolete("IDeleteOrganizationUserCommand should be used instead. To be removed by EC-607.")]
+    public async Task DeleteUserAsync(Guid organizationId, Guid organizationUserId,
+        EventSystemUser systemUser)
+    {
+        var orgUser = await RepositoryDeleteUserAsync(organizationId, organizationUserId, null);
+        await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Removed, systemUser);
+    }
+
+    private async Task<OrganizationUser> RepositoryDeleteUserAsync(Guid organizationId, Guid organizationUserId, Guid? deletingUserId)
```

Comment:
You can extract `deletingUserId` to the public method, similar to `invitingUserId`.

Reply:
I think each public method making the repository call is fine, or I'd have the public methods take the `OrganizationUser` object to begin with. But we've done lots of refactoring already so we can leave it if you disagree.